vehicles/src/modules/shopping-cart/shopping-cart.service.ts

```diff
@@ -0,0 +1,254 @@
+  ): Promise<Application[]> {
+    const { userGUID, orbcUserAuthGroup } = currentUser;
+    return await this.getSelectShoppingCartQB(
+      companyId,
+      userGUID,
+      orbcUserAuthGroup,
+    )
+      .orderBy({ 'application.updatedDateTime': 'DESC' })
+      .getMany();
+  }
+
+  /**
+   * Retrieves the count of permit applications currently in the shopping cart for a specific company.
+   * Optionally considers the user's GUID and authorization group for further filtering.
+   *
+   * @param currentUser - The current user's credentials and information.
+   * @param companyId - (Optional) The ID of the company for which to count permit applications in the shopping cart.
+   * @returns A promise resolved with the number of permit applications in the shopping cart that match the criteria.
+   */
+  @LogAsyncMethodExecution()
+  async getCartCount(
+    currentUser: IUserJWT,
+    companyId: number,
+  ): Promise<number> {
+    const { userGUID, orbcUserAuthGroup } = currentUser;
+    return await this.getSelectShoppingCartQB(
+      companyId,
+      userGUID,
+      orbcUserAuthGroup,
+    ).getCount();
+  }
+
+  /**
+   * Retrieves a `SelectQueryBuilder` for permit applications based on specified criteria.
+   * This method prepares a query to fetch applications with the status `IN_CART`, further filtered by company ID,
+   * and optionally by the user's GUID and authorization group for more fine-grained control.
+   *
+   * @param companyId - The ID of the company to filter permit applications by.
+   * @param userGUID - (Optional) The user's GUID to filter applications by, depending on the user's authorization group.
+   * @param orbcUserAuthGroup - (Optional) The user's authorization group which determines the level of access and filtering applied to the query.
+   * @returns A `SelectQueryBuilder` configured with the appropriate conditions to fetch the desired permit applications.
+   */
+  private getSelectShoppingCartQB(
+    companyId: number,
+    userGUID?: string,
+    orbcUserAuthGroup?: UserAuthGroup,
+  ): SelectQueryBuilder<Application> {
+    const queryBuilder = this.applicationRepository
+      .createQueryBuilder('application')
+      .leftJoin('application.company', 'company')
+      .leftJoin('application.applicationOwner', 'applicationOwner');
+
+    queryBuilder.where('application.permitStatus = :permitStatus', {
+      permitStatus: ApplicationStatus.IN_CART,
+    });
+    queryBuilder.andWhere('company.companyId = :companyId', { companyId });
+
+    // If user is a Permit Applicant, get only their own applications in cart
+    if (orbcUserAuthGroup === ClientUserAuthGroup.PERMIT_APPLICANT) {
+      queryBuilder.andWhere('applicationOwner.userGUID = :userGUID', {
+        userGUID,
+      });
+    }
+    // If user is a Company Admin, get all applications in cart for that company
+    // EXCEPT for those created by staff user.
+    else if (orbcUserAuthGroup === ClientUserAuthGroup.COMPANY_ADMINISTRATOR) {
```

Comment:
So when the Company Admin calls the endpoint that calls this method, this would return all of the applications in the cart (excluding the staff ones)? If so, then how to filter this from the FE ("All applications" or "My Applications")?

Reply:
For now, I'm sending applicantGUID in the response. Since there's no pagination involved, you'll get all the applications which you can use for the filtering.

I'm considering a query parameter but haven't come up with a proper design for it yet (should be available in a day or two).

Reply:
ok sure, np